src/pages/CoinDetails.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+import { useLocation, Navigate } from 'react-router-dom';
+import './CoinDetails.css';
+
+const CoinDetails = () => {
+  const location = useLocation();
+  useEffect(() => {
+    window.scrollTo(20, 0);
+  });
+
+  const singleData = location?.state?.coin;
+  if (!singleData) {
+    return <Navigate to="/" replace />;
+  }
+  return (
+    <div>
+      <div className="container">
+        <p>{`Coins supply in market: ${singleData.csupply}`}</p>
+      </div>
+      <p className="name">{singleData.name}</p>
+      <p className="name">{`Rank:  ${singleData.rank}`}</p>
+      <p className="name">{`Symbol:  ${singleData.symbol}`}</p>
+      <p className="name">{`Market cap in USD:  $${singleData.market_cap_usd}`}</p>
+      <p className="name">{`Weekly percentage change:  ${singleData.percent_change_7d}`}</p>
+      <p className="name">{`Price:  $${singleData.price_usd}`}</p>
+    </div>
+  );
+};
```

Comment:
- [ ] Kindly make sure your `Detail` pages design follow the one given from the template like **the height of each detail row**, **alignment of text element(to have space-between)**,...

| Template Design | Your Design |
| --------------------- | ------------------------- |
| ![row_detail](https://user-images.githubusercontent.com/83042742/188026547-407d5325-a1f8-4291-a698-43a3b9520f40.png) | ![Screenshot from 2022-09-02 00-54-08](https://user-images.githubusercontent.com/83042742/188026571-29f7cb3f-6231-467a-af5f-04b320c7170a.png) |